### PR TITLE
Enable integration-test workflow for PRs

### PR DESCRIPTION
**Pull request by jcheng-splunk**

### CODE REVIEW

File: .github/workflows/integration-test.yml [hunk 1]
```diff
@@ -4,6 +4,20 @@ on:
...
+      - 'go.mod'
+      - 'go.sum'
+      - 'cmd/otelcol/**'
+      - 'internal/**'
+      - '!internal/buildscripts/packaging/choco/**'
+      - '!internal/buildscripts/packaging/fpm/**'
+      - '!internal/buildscripts/packaging/installer/**'
+      - '!internal/buildscripts/packaging/msi/**'
+      - '!internal/buildscripts/packaging/release/**'
+      - '!internal/buildscripts/packaging/tests/**'
```
Comment (rmfitzpatrick) [inline .github/workflows/integration-test.yml:19]: Would a more general `'!internal/buildscripts/**'` be wanted? I'm not sure if `update-deps` is worth running these for instance.

Comment (jcheng-splunk) [reply .github/workflows/integration-test.yml:19]: Unfortunately, I think we need to include `internal/buildscripts/packaging/collect-libs.sh` and `internal/buildscripts/packaging/smart-agent-release.txt` for the docker image build.  We can move these files to another dir and update the build scripts in a future PR?

Comment (rmfitzpatrick) [reply .github/workflows/integration-test.yml:19]: More a question than a suggestion, but if we find this runs unnecessarily we can update as you suggest.

**Review by rmfitzpatrick — APPROVED** [review]
🤞

File: .github/workflows/integration-test.yml (hunk 1 above)
Comment (jcheng-splunk) [reply .github/workflows/integration-test.yml:19]: Maybe we can just use:
```suggestion
      - '!internal/buildscripts/*'
      - '!internal/buildscripts/packaging/*/**'
```
This should ignore everything currently in `internal/buildscripts/` except those 2 files.

**Pull request merged**